src/targets/gpu/mlir.cpp

```diff
@@ -614,9 +614,16 @@ struct mlir_program
         return "migraphx." + ins->name();
     }
 
-    static value get_operator_value(const operation& op)
+    static value get_operator_value(instruction_ref ins)
     {
-        auto v = op.to_value();
+        const operation& op = ins->get_operator();
+        auto v              = op.to_value();
+
+        // Reshape operator can have dim 0 or -1.
```

Comment:
@CharlieL7  do you know if `dims` can be normalized using `normalize_attributes` somehow ?

Reply:
> dim 0 is it allowed ?

Yes.

https://github.com/ROCm/AMDMIGraphX/blob/3409e476f616bd19f54017beeac59b7e56f273b5/src/include/migraphx/op/reshape.hpp#L117

Reply:
> We could normalize the dims attribute in reshape for the static shape case by adding some modes to normalize_attributes. 

It may be better to add, i am not sure how much complicated it would be. 

also existing approch in this PR would affect the compile time so dynamic shapes are not handled